storage/repository/parse.go

```diff
@@ -60,13 +61,16 @@ func ParseURI(s string) (fyne.URI, error) {
 		// we should punt this to whoever generated the URI in the
 		// first place?
 
-		path := s[5:] // everything after file:
-		if len(path) > 2 && path[:2] == "//" {
-			path = path[2:]
-		}
+		if len(s) > 5 {
+			path := s[5:] // everything after file:
+			if len(path) > 2 && path[:2] == "//" {
+				path = path[2:]
+			}
 
-		// Windows files can break authority checks, so just return the parsed file URI
-		return NewFileURI(path), nil
+			// Windows files can break authority checks, so just return the parsed file URI
+			return NewFileURI(path), nil
+		}
+		return nil, errors.New("not a valid URI")
```

Comment:
I would prefer if you check that the code path is invalid and error out in this case. In other words, you keep the code indented as it was before and then have the error handing in an if-statement instead.
```go
if len(s) < 6 {
   return nil, errors.New("not a valid URI")
}
```